### PR TITLE
8231500: [lworld] Merge the experimental bytecode API

### PR DESCRIPTION
**Pull request by MrSimms**
* Moved to a single bytecode testing API
* Cleaned out vbytecodes
* Cleanup of tests not actually using the API
* Removed TestNativeClone's API usage since cooking the method handle could no
longer bypass the verifier's bad access check

---------
### Progress
- [x] Change must not contain extraneous whitespace

### Issue
 * [JDK-8231500](https://bugs.openjdk.java.net/browse/JDK-8231500): [lworld] Merge the experimental bytecode API

### Download
`$ git fetch https://git.openjdk.java.net/valhalla pull/279/head:pull/279`
`$ git checkout pull/279`

### CODE REVIEW

File: test/jdk/java/lang/invoke/common/test/java/lang/invoke/lib/InstructionHelper.java [hunk 1]
```diff
@@ -148,4 +159,199 @@ public static String csym(Class<?> c) {
...
+            public String type(Class<?> aClass) {
+                if (aClass.isArray()) {
+                    return classToInternalName(aClass);
+                } else {
+                    return (aClass.isInlineClass() ? "Q" : "L") + classToInternalName(aClass) + ";";
+                }
+            }
+
+            @Override
+            public boolean isValue(String desc) {
```
Comment (mlchung) [inline test/jdk/java/lang/invoke/common/test/java/lang/invoke/lib/InstructionHelper.java:271]: I think we should rename `isValue` to `isInlineClass` and also other new `xxxValue` methods

File: test/jdk/java/lang/invoke/common/test/java/lang/invoke/lib/InstructionHelper.java [hunk 2]
```diff
@@ -148,4 +159,199 @@ public static String csym(Class<?> c) {
...
+
+
+    // loadCode(MethodHandles.Lookup, String, MethodType, Consumer<? super MethodHandleCodeBuilder<?>>) et al...
+
+    public static MethodHandle loadCode(MethodHandles.Lookup lookup, String name, MethodType type, Consumer<? super MethodHandleCodeBuilder<?>> builder) {
+        String className = generateClassNameFromLookupClass(lookup);
+        return loadCode(lookup, className, name, type, builder);
+    }
+
+    public static MethodHandle loadCode(MethodHandles.Lookup lookup, String className, String methodName, MethodType type, Consumer<? super MethodHandleCodeBuilder<?>> builder) {
```
Comment (mlchung) [inline test/jdk/java/lang/invoke/common/test/java/lang/invoke/lib/InstructionHelper.java:171]: Are these two `loadCode` methods taking a builder `Consumer` used by the tests?

If so, the following classes and methods can be made private to hide them as internals.

File: test/jdk/lib/testlibrary/bytecode/jdk/experimental/bytecode/BasicTypeHelper.java [hunk 3]
```diff
@@ -45,16 +45,27 @@ public String arrayOf(String s) {
         return "[" + s;
     }
 
+    @Override
+    public boolean isValue(String t) {
```
Comment (mlchung) [inline test/jdk/lib/testlibrary/bytecode/jdk/experimental/bytecode/BasicTypeHelper.java:49]: s/isValue/isInlineClass/

File: test/jdk/lib/testlibrary/bytecode/jdk/experimental/bytecode/BasicTypeHelper.java [hunk 4]
```diff
@@ -45,16 +45,27 @@ public String arrayOf(String s) {
...
+        return t.charAt(0) == 'Q' && t.endsWith(";");
+    }
+
     @Override
     public String type(String s) {
         return "L" + s + ";";
     }
 
+    @Override
+    public String valueType(String s) {
```
Comment (mlchung) [inline test/jdk/lib/testlibrary/bytecode/jdk/experimental/bytecode/BasicTypeHelper.java:59]: s/valueType/inlineType/

File: test/jdk/lib/testlibrary/bytecode/jdk/experimental/bytecode/BytePoolHelper.java [hunk 5]
```diff
@@ -336,6 +337,11 @@ public int putClass(S symbol) {
         return putClassInternal(symbolToString.apply(symbol));
     }
 
+    @Override
+    public int putValueClass(S symbol) {
```
Comment (mlchung) [inline test/jdk/lib/testlibrary/bytecode/jdk/experimental/bytecode/BytePoolHelper.java:341]: s/putValueClass/putInlineClass/

**Review by mlchung — COMMENTED** [review]
Looks okay in general.   For the new methods like `isValue`, `valueType` and `putValueClass` etc, they should be renamed to reflect the new terminology (inline class).

File: test/jdk/java/lang/invoke/common/test/java/lang/invoke/lib/InstructionHelper.java (hunk 1 above)
Comment (MrSimms) [reply test/jdk/java/lang/invoke/common/test/java/lang/invoke/lib/InstructionHelper.java:271]: Comments applied

File: test/jdk/java/lang/invoke/common/test/java/lang/invoke/lib/InstructionHelper.java (hunk 2 above)
Comment (MrSimms) [reply test/jdk/java/lang/invoke/common/test/java/lang/invoke/lib/InstructionHelper.java:171]: Comments applied

File: test/jdk/lib/testlibrary/bytecode/jdk/experimental/bytecode/BasicTypeHelper.java (hunk 3 above)
Comment (MrSimms) [reply test/jdk/lib/testlibrary/bytecode/jdk/experimental/bytecode/BasicTypeHelper.java:49]: Comments applied

File: test/jdk/lib/testlibrary/bytecode/jdk/experimental/bytecode/BasicTypeHelper.java (hunk 4 above)
Comment (MrSimms) [reply test/jdk/lib/testlibrary/bytecode/jdk/experimental/bytecode/BasicTypeHelper.java:59]: Comments applied

File: test/jdk/lib/testlibrary/bytecode/jdk/experimental/bytecode/BytePoolHelper.java (hunk 5 above)
Comment (MrSimms) [reply test/jdk/lib/testlibrary/bytecode/jdk/experimental/bytecode/BytePoolHelper.java:341]: Comments applied

**Pull request closed without merging**